### PR TITLE
[feat]: [CDS-38924]: AWS IAM+IRSA support for serverless lambda

### PR DESCRIPTION
**Pull request by PragyeshMishra01**
## Describe your changes

For serverless Lambda swimlane- We are providing support to have an AWS connector with "assume IAM Role" and "Use IRSA" credential type.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34400)

### CODE REVIEW

**Comment by PragyeshMishra01** [comment]
trigger messagecheck

**Comment by PragyeshMishra01** [comment]
trigger recency

**Comment by PragyeshMishra01** [comment]
trigger messagecheck

**Comment by PragyeshMishra01** [comment]
trigger ti0

**Comment by PragyeshMishra01** [comment]
trigger ti2

File: 930-delegate-tasks/src/main/java/io/harness/delegate/serverless/ServerlessAwsLambdaPrepareRollbackCommandTaskHandler.java [hunk 1]
```diff
@@ -91,9 +95,12 @@ protected ServerlessCommandResponse executeTaskInternal(ServerlessCommandRequest
           LogLevel.ERROR, CommandExecutionStatus.FAILURE);
       throw ex;
     }
+    serverlessClient = ServerlessClient.client(serverlessDelegateTaskParams.getServerlessClientPath());
 
     try {
-      configureCredential(serverlessPrepareRollbackDataRequest, executionLogCallback, serverlessDelegateTaskParams);
+      if (serverlessAwsLambdaCredentialType.equals(AwsCredentialType.MANUAL_CREDENTIALS.name())) {
```
Comment (sainath-b) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/serverless/ServerlessAwsLambdaPrepareRollbackCommandTaskHandler.java:101]: duplicate code in deploy command and rollback task handler can we move to some common place

Comment (PragyeshMishra01) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/serverless/ServerlessAwsLambdaPrepareRollbackCommandTaskHandler.java]: done

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 2 of 5 files at r1, 3 of 3 files at r3, 3 of 4 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34400)* status: :shipit: complete! ✓

**Pull request merged**